File: CS/dxSample_ButtonColoredStates/ImageStateHelper.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using DevExpress.LookAndFeel;
using DevExpress.Skins;

namespace dxSample_ButtonColoredStates {
    public static class ImageStateHelper {
        public static void ColorButtonState(int stateNumber, System.Drawing.Color stateColor) {
            SetColoredState(stateNumber, stateColor);
        }
        private static void SetColoredState(int stateNumber, Color stateColor) {
            var skin = GetCurrentSkin();
            var element = skin["Button"];
            var initialImg = element.Image;
            if (initialImg == null) {
                return;
            }
            var images = initialImg.GetImages().Images;
            if (images == null || images.Count == 0) {
                return;
            }
            var imgForState = images[stateNumber];
            var imgSize = imgForState.Size;
            var g = Graphics.FromImage(element.Image.Image);
            g.FillRectangle(new SolidBrush(stateColor), new Rectangle(new Point(0, stateNumber * imgSize.Height), imgSize));
        }
        private static Skin GetCurrentSkin() {
            return CommonSkins.GetSkin(UserLookAndFeel.Default.ActiveLookAndFeel);
        }
    }
}

File: VB/dxSample_ButtonColoredStates/ImageStateHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports DevExpress.LookAndFeel
Imports DevExpress.Skins

Namespace dxSample_ButtonColoredStates
    Public NotInheritable Class ImageStateHelper

        Private Sub New()
        End Sub

        Public Shared Sub ColorButtonState(ByVal stateNumber As Integer, ByVal stateColor As System.Drawing.Color)
            SetColoredState(stateNumber, stateColor)
        End Sub
        Private Shared Sub SetColoredState(ByVal stateNumber As Integer, ByVal stateColor As Color)
            Dim skin = GetCurrentSkin()
            Dim element = skin("Button")
            Dim initialImg = element.Image
            If initialImg Is Nothing Then
                Return
            End If
            Dim images = initialImg.GetImages().Images
            If images Is Nothing OrElse images.Count = 0 Then
                Return
            End If
            Dim imgForState = images(stateNumber)
            Dim imgSize = imgForState.Size
            Dim g = Graphics.FromImage(element.Image.Image)
            g.FillRectangle(New SolidBrush(stateColor), New Rectangle(New Point(0, stateNumber * imgSize.Height), imgSize))
        End Sub
        Private Shared Function GetCurrentSkin() As Skin
            Return CommonSkins.GetSkin(UserLookAndFeel.Default.ActiveLookAndFeel)
        End Function
    End Class
End Namespace
